core/src/test/scala/codesearch/core/regex/space/SpaceInsensitiveSpec.scala

```diff
@@ -0,0 +1,43 @@
+package codesearch.core.regex.space
+
+import org.scalatest.{FreeSpec, Matchers}
+
+class SpaceInsensitiveSpec extends FreeSpec with Matchers {
+  def addingSpaceInsensitive(before: String, after: String) = {
+    before in {
+      SpaceInsensitive.spaceInsensitiveString(before) shouldBe after
+    }
+  }
+
+  "Space Insensitive tests" - {
+    "Simple strings" - {
+      addingSpaceInsensitive("Hello world", "Hello +world")
+
+      addingSpaceInsensitive("Hello    world", "Hello    world")
+    }
+
+    "Strings with regexp" - {
+      addingSpaceInsensitive("foo +bar", "foo +bar")
+
+      addingSpaceInsensitive("foo ?bar", "foo( +)?bar")
+
+      addingSpaceInsensitive("foo *bar", "foo *bar")
+    }
+
+    "Strings with other cases" - {
+      addingSpaceInsensitive("foo [bar]", "foo +[bar]")
+
+      addingSpaceInsensitive("foo {bar}", "foo {bar}")
```

Comment:
A vot i net, this should actually become `foo +{bar}` because not everything in curly braces is parsed as a range. You can leave this case as-is, just add a comment that this is actually a bug.

On the other hand, you _do_ need a test like `addingSpaceInsensitive("foo {1,3}", "foo {1,3}")`.